### PR TITLE
Add various bounds checks

### PR DESCRIPTION
**Pull request by jtbandes**
Add bounds checking for Table construction and readScalar.
Fixes https://github.com/jkuszmaul/flatbuffers_reflection/issues/46, maybe?

Relates to https://github.com/foxglove/studio/issues/5314
cc @snosenzo

### CODE REVIEW

**Comment by jkuszmaul** [comment]
To a first approximation, this looks reasonable and I think addresses the core issue (which is more about catching grossly misserialized messages rather than trying to catch literally every single possible corner case).

Still need to review the exact changes to double-check that the exact checks are correct and to see if any other test cases should get added.

File: src/reflection.ts [hunk 1]
```diff
@@ -151,6 +202,12 @@ export class Table {
   }
   // Reads a scalar of a given type at a given offset.
   readScalar(fieldType: reflection.BaseType, offset: number): number | BigInt {
+    const size = typeSize(fieldType);
+    if (offset + size > this.bb.capacity()) {
```
Comment (jkuszmaul) [inline src/reflection.ts:206]: Also check that `offset > 0`.

**Review by jkuszmaul — COMMENTED** [review]
Logic LGTM and thank you for the tests.

**Review by snosenzo — APPROVED** [review]

**Review by jkuszmaul — APPROVED** [review]
